### PR TITLE
Build envelopes on request.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,10 +2,171 @@ name: Build
 
 on:
   workflow_dispatch:
+    inputs:
+      toit-version:
+        description: Toit SDK version to check out
+        required: true
+        type: string
+      upload-release:
+        description: Whether to upload release artifacts
```
Comment (kasperl) [inline .github/workflows/build.yml:11]:
```suggestion
        description: Upload release artifacts
```

File: tools/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+build the host SDK into `build/host/sdk`.
+
+The `toit.run` and `toit.pkg` are then in `build/host/sdk/bin`.
+
+In the remainder of this document, we assume that you have added
+`build/host/sdk/bin` to your `PATH`. If not, just replace `toit.run`
+and `toit.pkg` with the full path to the binaries.
+
+## The envelope tool
+
```
Comment (kasperl) [inline tools/README.md:26]: Drop the newlines after # for consistency?

File: README.md [hunk 3]
```diff
@@ -1 +1,68 @@
 # Envelopes
+
```
Comment (kasperl) [inline README.md:2]: Consider dropping the newlines after # for consistency.

File: tools/main.toit [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+  return "toolchains/$chip/sdkconfig.defaults"
+toit_main_path_for_ --chip/string -> string:
+  return "toolchains/$chip/main"
+
+main args:
+  ui := Ui_
+  root_cmd := cli.Command "root"
+
+  variant_list_cmd := cli.Command "list"
+      --short_help="Lists all variants."
```
Comment (kasperl) [inline tools/main.toit:29]:
```suggestion
      --short_help="List all variants."
```

File: tools/main.toit [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        cli.Option "root"
+            --short_help="The root directory to list variants from."
+            --default="variants",
+      ]
+      --run=:: variant_list it --ui=ui
+  root_cmd.add variant_list_cmd
+
+  variant_synthesize_cmd := cli.Command "synthesize"
+      --long_help="""
+        Synthesizes the given variants.
```
Comment (kasperl) [inline tools/main.toit:40]:
```suggestion
        Synthesize the given variants.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: README.md (hunk 3 above)
Comment (floitsch) [reply README.md:2]: Apparently the preferred way to write markdown is with new-lines.
https://www.markdownguide.org/basic-syntax/#heading-best-practices

A lot of our docs have been written without them, but I tried to be consistent here.

File: tools/README.md (hunk 2 above)
Comment (floitsch) [reply tools/README.md:27]: Added them for consistency. (See the other comment and https://www.markdownguide.org/basic-syntax/#heading-best-practices)

**Pull request merged**